left-right/src/write.rs

```diff
@@ -299,18 +299,7 @@ where
     ///
     /// Its effects will not be exposed to readers until you call [`publish`](Self::publish).
     pub fn append(&mut self, op: O) -> &mut Self {
-        if self.first {
-            // Safety: we know there are no outstanding w_handle readers, since we haven't
-            // refreshed ever before, so we can modify it directly!
-            let mut w_inner = self.raw_write_handle();
-            let w_inner = unsafe { w_inner.as_mut() };
-            let r_handle = self.enter().expect("map has not yet been destroyed");
-            // Because we are operating directly on the map, and nothing is aliased, we do want
-            // to perform drops, so we invoke absorb_second.
-            Absorb::absorb_second(w_inner, op, &*r_handle);
-        } else {
-            self.oplog.push_back(op);
-        }
+        self.extend(vec![op]);
```

Comment:
I think this should probably use `std::iter::once` so we don't have to allocate a `Vec`.

Reply:
ah, yes, I knew something like that had to exist!